indy_node/server/client_authn.py

```diff
@@ -39,9 +41,31 @@ def serializeForSig(self, msg, identifier=None, topLevelKeysToIgnore=None):
     def addIdr(self, identifier, verkey, role=None):
         raise RuntimeError('Add verification keys through the NYM txn')
 
-    def getVerkey(self, identifier):
+    def getVerkey(self, ident, request):
         try:
-            verkey = self.cache.getVerkey(identifier, isCommitted=False)
+            verkey = self.cache.getVerkey(ident, isCommitted=False)
         except KeyError:
+            # If DID wasn't found in idr cache, it might be
+            # non-ledger request, so we need to look for verkey in request
```

Comment:
A more clear comment:
`If DID wasn't found in idr cache, then it can be creation of a new DID signed by a verkey for this new DID`.

Reply:
ok